tgui/packages/tgui/interfaces/NtosMessenger.tsx

```diff
@@ -0,0 +1,469 @@
+    this.handleMessageInput = this.handleMessageInput.bind(this);
+    this.handleSendMessage = this.handleSendMessage.bind(this);
+  }
+
+  componentDidMount(): void {
+    this.scrollToBottom();
+  }
+
+  componentDidUpdate(
+    prevProps: ChatScreenProps,
+    _prevState: ChatScreenState,
+    _snapshot: any
+  ): void {
+    if (prevProps.msgs.length !== this.props.msgs.length) {
+      this.scrollToBottom();
+    }
+  }
+
+  scrollToBottom(): void {
+    const scroll = this.scrollRef.current;
+    if (scroll !== null) {
+      scroll.scrollTop = scroll.scrollHeight;
+    }
+  }
+
+  handleSendMessage(): void {
+    if (this.state.msg === '') return;
+    const { act } = useBackend<NtosMessengerData>(this.context);
+    act('PDA_sendMessage', {
+      ref: this.props.recp.ref,
+      name: this.props.recp.name,
+      job: this.props.recp.job,
+      msg: this.state.msg,
+    });
+    this.setState({ msg: '', canSend: false });
+    setTimeout(() => this.setState({ canSend: true }), 1000);
+  }
+
+  handleMessageInput(_: any, val: string): void {
+    this.setState({ msg: val });
+  }
+
+  render(): InfernoNode | undefined {
+    const { act } = useBackend<NtosMessengerData>(this.context);
+    const { recp, onReturn, msgs } = this.props;
+    const { msg, canSend } = this.state;
+
+    let lastMsgRef = '';
+    let filteredMessages: JSX.Element[] = [];
+
+    for (let index = 0; index < msgs.length; index++) {
+      let message = msgs[index];
+
+      const isSwitch = lastMsgRef !== message.sender;
+      lastMsgRef = message.sender;
+
+      filteredMessages.push(
+        <Stack.Item key={index} mt={isSwitch ? 2 : 0.5}>
+          <ChatMessage
+            isSelf={message.outgoing}
+            msg={message.contents}
+            everyone={!!message.everyone}
+            photoPath={message.photo_path}
+          />
+        </Stack.Item>
+      );
+    }
+
+    return (
+      <Stack vertical fill>
+        <Stack.Item>
+          <Section fill>
+            <Button icon="arrow-left" content="Back" onClick={onReturn} />
+            <Button.Confirm
+              icon="trash-can"
+              content="Delete chat"
+              onClick={() => act('PDA_clearMessages', { ref: recp.ref })}
+            />
+          </Section>
+        </Stack.Item>
+
+        <Stack.Item grow={1}>
+          <Section
+            scrollable
+            fill
+            fitted
+            title={`${recp.name} (${recp.job})`}
+            scrollableRef={this.scrollRef}>
+            <Stack vertical fill className="NtosMessenger__ChatLog">
+              <Stack.Item textAlign="center" fontSize={1}>
+                This is the beginning of your chat with {recp.name}.
+              </Stack.Item>
+              <Stack.Divider />
+              {filteredMessages}
+            </Stack>
+          </Section>
+        </Stack.Item>
+
+        <Stack.Item>
+          <Section>
+            <Stack fill>
+              <Stack.Item grow={1}>
```

Comment:
minor nit, here and in other places, when you're using grow  1, simply typing `grow` is fine